elements/ip/generateipfilterrules.cc

```diff
@@ -154,30 +154,30 @@ GenerateIPFilter::configure(Vector<String> &conf, ErrorHandler *errh)
             new IPClassifierRuleFormatter(_keep_sport, _keep_dport));
     }
 
-    return GenerateIPPacket::configure(conf, errh);
+    return GenerateIPPacketRules::configure(conf, errh);
 }
 
 int
-GenerateIPFilter::initialize(ErrorHandler *errh)
+GenerateIPFilterRules::initialize(ErrorHandler *errh)
 {
-    return GenerateIPPacket::initialize(errh);
+    return GenerateIPPacketRules::initialize(errh);
 }
 
 void
-GenerateIPFilter::cleanup(CleanupStage)
+GenerateIPFilterRules::cleanup(CleanupStage)
 {
 }
 
 Packet *
-GenerateIPFilter::simple_action(Packet *p)
+GenerateIPFilterRules::simple_action(Packet *p)
 {
     // Create a flow signature for this packet
     IPFlowID flowid(p);
     IPFlow new_flow = IPFlow();
     new_flow.initialize(flowid & _mask);
 
     // Check if we already have such a flow
-    IPFlow *found = _map.find(flowid).get();
+    IPFlow *found = _map.find(flowid & _mask).get();
```

Comment:
Don'y you think that this was a bug?
Line 177 initializes a flow with a flow ID which derives from flowid & mask, but then line 180 searches for an existing flow with flowid. Apparently this is not the same flow

Reply:
What is sure is that new_flow should be in the if branch. And then yes we need & _mask in both side. Probably flowid = flowid & _mask as we won't use it unmasked.